### PR TITLE
[Don't merge] lp1748758: Fix metadata export of selected tracks

### PR DESCRIPTION
**Pull request by uklotzde**
https://bugs.launchpad.net/mixxx/+bug/1748758

- Metadata must be exported for selected tracks even if the general preference option is disabled
- The _dirty_ and the _export metadata_ flag of a track need to be handled separately

### CODE REVIEW

File: src/library/dao/trackdao.cpp [hunk 1]
```diff
@@ -191,40 +191,46 @@ bool TrackDAO::trackExistsInDatabase(const QString& absoluteFilePath) {
...
+    // Write audio meta data, if explicitly requested by the user
+    // for individual tracks or enabled in the preferences for all
+    // tracks.
+    //
+    // This must be done before updating the database, because
+    // a timestamp is used to keep track of when metadata has been
+    // last synchronized. Exporting metadata will update this time
+    // stamp on the track object!
+    if (pTrack->isMarkedForMetadataExport() ||
+            (pTrack->isDirty() && m_pConfig && m_pConfig->getValueString(ConfigKey("[Library]","SyncTrackMetadataExport")).toInt() == 1)) {
```
Comment (daschuer) [inline src/library/dao/trackdao.cpp]: This line is too long for GitHub code review ;-)

File: src/track/track.cpp [hunk 2]
```diff
@@ -830,9 +830,14 @@ bool Track::isDirty() {
 
 void Track::markForMetadataExport() {
     QMutexLocker lock(&m_qMutex);
-    if (compareAndSet(&m_bMarkedForMetadataExport, true)) {
-        markDirtyAndUnlock(&lock);
-    }
+    m_bMarkedForMetadataExport = true;
```
Comment (daschuer) [inline src/track/track.cpp]: I think we should also unset this flag after export. It does not cost much, and we do not need to worry about it if we discuss the changes in the concurrent cache PR. 
A similar issue would be to re-allow signals from the track.

**Comment by uklotzde** [comment]
Merged into #1492 to prevent non-trivial merge conflicts

**Comment by uklotzde** [comment]
Doesn't need to be merged, because it will be superseded by #1492 as already mentioned. I will close this PR after #1492 has been merged.

**Pull request merged**